OnlinerTask/OnlinerTask.Data/Repository/Implementations/MsSQLProductRepository.cs

```diff
@@ -25,32 +22,63 @@ public MsSqlProductRepository(IRepository repository)
             return products;
         }
 
-        public async Task<bool> CheckItem(int itemId, string username)
+        public UsersUpdateEmail WriteUpdate(ProductModel item, string useremail)
         {
-            using (var context = new OnlinerProducts())
+            using (var db = new ApplicationDbContext())
             {
-                return await context.Product.FirstOrDefaultAsync(x => x.ProductId == itemId && x.UserEmail == username) != null;
+                var time = db.Users.FirstOrDefault(x => x.Email == useremail);
+                if (time != null)
+                {
+                    return WriteUpdateToProduct(item, time.EmailTime);
+                }
             }
+            return null;
         }
 
-        public List<Product> GetAllProducts()
+        private async Task<bool> CheckItem(int itemId, string username)
         {
-            return repository.GetAllProducts();
-        }
-
-        public List<Product> GetPersonalProducts(string name)
-        {
-            return repository.GetPersonalProducts(name);
+            using (var context = new OnlinerProducts())
+            {
+                return await context.Product.FirstOrDefaultAsync(x => x.ProductId == itemId && x.UserEmail == username) != null;
+            }
         }
 
-        public Task<string> RemoveOnlinerProduct(int itemId, string name)
+        private void UpdateProduct(Product item, ProductModel model)
         {
-            return repository.RemoveOnlinerProduct(itemId, name);
+            if (item == null)
+            {
+                return;
+            }
+            using (var db = new OnlinerProducts())
+            {
+                var product = db.Product.FirstOrDefault(x => x.ProductId == item.ProductId);
+                if (product == null)
+                {
+                    return;
+                }
+                product.Price.PriceMaxAmmount.Amount = model.Prices.PriceMax.Amount;
+                product.Price.PriceMinAmmount.Amount = model.Prices.PriceMin.Amount;
+                db.SaveChanges();
+            }
         }
 
-        public bool CreateOnlinerProduct(ProductModel model, string userEmail)
+        private UsersUpdateEmail WriteUpdateToProduct(ProductModel model, TimeSpan time)
         {
-            return repository.CreateOnlinerProduct(model, userEmail);
+            if (model == null)
+            {
+                return null;
+            }
+            using (var db = new OnlinerProducts())
```

Comment:
Dont like new statement for implementations. Lets create facroty for contexts or simple injection.